tests/datasets/test_utils.py

```diff
@@ -306,26 +445,103 @@ def test_disambiguate_timestamp(
-        sample["bar"], torch.tensor([2, 4])  # type: ignore[attr-defined]
-    )
+class TestCollateFunctionsMatchingKeys:
+    @pytest.fixture(scope="class")
+    def samples(self) -> List[Dict[str, Any]]:
+        return [
+            {
+                "image": torch.tensor([1, 2, 0]),  # type: ignore[attr-defined]
+                "crs": CRS.from_epsg(2000),
+            },
+            {
+                "image": torch.tensor([0, 0, 3]),  # type: ignore[attr-defined]
+                "crs": CRS.from_epsg(2001),
+            },
+        ]
+
+    def test_stack_samples(self, samples: List[Dict[str, Any]]) -> None:
+        sample = stack_samples(samples)
+        assert sample["image"].size() == torch.Size(  # type: ignore[attr-defined]
+            [2, 3]
+        )
+        assert torch.allclose(  # type: ignore[attr-defined]
+            sample["image"],
+            torch.tensor([[1, 2, 0], [0, 0, 3]]),  # type: ignore[attr-defined]
+        )
+        assert sample["crs"] == [CRS.from_epsg(2000), CRS.from_epsg(2001)]
+
+    def test_concat_samples(self, samples: List[Dict[str, Any]]) -> None:
+        sample = concat_samples(samples)
+        assert sample["image"].size() == torch.Size([6])  # type: ignore[attr-defined]
+        assert torch.allclose(  # type: ignore[attr-defined]
+            sample["image"],
+            torch.tensor([1, 2, 0, 0, 0, 3]),  # type: ignore[attr-defined]
+        )
+        assert sample["crs"] == CRS.from_epsg(2000)
+
+    def test_merge_samples(self, samples: List[Dict[str, Any]]) -> None:
+        sample = merge_samples(samples)
+        assert sample["image"].size() == torch.Size([3])  # type: ignore[attr-defined]
+        assert torch.allclose(  # type: ignore[attr-defined]
+            sample["image"], torch.tensor([1, 2, 3])  # type: ignore[attr-defined]
+        )
+        assert sample["crs"] == CRS.from_epsg(2001)
+
+
+class TestCollateFunctionsDifferingKeys:
+    @pytest.fixture(scope="class")
+    def samples(self) -> List[Dict[str, Any]]:
+        return [
+            {
+                "image": torch.tensor([1, 2, 0]),  # type: ignore[attr-defined]
+                "crs1": CRS.from_epsg(2000),
+            },
+            {
+                "mask": torch.tensor([0, 0, 3]),  # type: ignore[attr-defined]
+                "crs2": CRS.from_epsg(2001),
+            },
+        ]
+
+    def test_stack_samples(self, samples: List[Dict[str, Any]]) -> None:
+        sample = stack_samples(samples)
+        assert sample["image"].size() == torch.Size(  # type: ignore[attr-defined]
+            [1, 3]
+        )
+        assert sample["mask"].size() == torch.Size([1, 3])  # type: ignore[attr-defined]
+        assert torch.allclose(  # type: ignore[attr-defined]
+            sample["image"], torch.tensor([[1, 2, 0]])  # type: ignore[attr-defined]
+        )
+        assert torch.allclose(  # type: ignore[attr-defined]
+            sample["mask"], torch.tensor([[0, 0, 3]])  # type: ignore[attr-defined]
```

Comment:
At some point I'm going to go insane and add type hints for all of PyTorch because these `# type: ignore` comments are getting out of hand...

Reply:
Surely this is a problem for other people too?